File: benchmarks/map_operations/maps.js
const N = 100000;

const m = new Map();
for (let i = 0; i < N; i++) {
  m.set("key_" + i, i * 2);
}
console.log(m.size);

let found = 0;
for (let i = 0; i < N; i++) {
  if (m.has("key_" + i)) found++;
}
console.log(found);

let sum = 0;
for (const v of m.values()) {
  sum += v;
}
console.log(sum);

for (let i = 0; i < N; i += 2) {
  m.delete("key_" + i);
}
console.log(m.size);

File: benchmarks/map_operations/maps.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	N := 100000

	m := make(map[string]int)
	for i := 0; i < N; i++ {
		m["key_"+strconv.Itoa(i)] = i * 2
	}
	fmt.Println(len(m))

	found := 0
	for i := 0; i < N; i++ {
		if _, ok := m["key_"+strconv.Itoa(i)]; ok {
			found++
		}
	}
	fmt.Println(found)

	sum := 0
	for _, v := range m {
		sum += v
	}
	fmt.Println(sum)

	for i := 0; i < N; i += 2 {
		delete(m, "key_"+strconv.Itoa(i))
	}
	fmt.Println(len(m))
}
